module/state_synchronization/indexer/indexer_core.go

```diff
@@ -260,3 +276,57 @@ func (c *IndexerCore) indexRegisters(registers map[ledger.Path]*ledger.Payload,
 
 	return c.registers.Store(regEntries, height)
 }
+
+func (c *IndexerCore) trackExecutedMetricForCollection(light flow.LightCollection) {
```

Comment:
how about refactoring this a bit more to put these fields in their own struct (`accessMetrics`,  `collectionsToMarkFinalized`, `collectionsToMarkExecuted`), then adding methods for each of the `track*` methods.

that way we keep all the logic together